src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,199 @@
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE cars.id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(query)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+            if (updateStatement.executeUpdate() > 0) {
+                return car;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car " + car, e);
+        }
+        deleteDriversFromCars_Drivers(car.getId());
+        insertDrivers(car);
+        throw new NoSuchElementException("car does`t exist in DB");
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return (statement.executeUpdate() > 0);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete data by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT cars.id, model, manufacturer_id,manufacturers.id,"
+                + " manufacturers.name,manufacturers.country  "
+                + "FROM cars"
+                + " LEFT JOIN cars_drivers ON cars_id = cars.id "
+                + " LEFT JOIN manufacturers ON cars.manufacturer_id = manufacturers.id"
+                + " WHERE drivers_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getIdCarStatement = connection.prepareStatement(query)) {
+            getIdCarStatement.setLong(1, driverId);
+            ResultSet resultSet = getIdCarStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all by driver " + driverId, e);
+        }
+        // get info about drivers for each car
+        return carList;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject(1, Long.class));
+        car.setModel(resultSet.getString(2));
+        car.setManufacturer(new Manufacturer(resultSet.getObject(4, Long.class),
+                resultSet.getString(5), resultSet.getString(6)));
+        car.setDriver(getDriversFromCars_drivers(resultSet.getLong(1)));
```

Comment:
move this logic out of this method
![image](https://user-images.githubusercontent.com/20326858/223412948-8e204ca7-b2a7-4953-b9ff-29cf10b105bc.png)